web/pgadmin/static/js/PgTreeView/index.jsx

```diff
@@ -38,9 +38,11 @@ const Root = styled('div')(({theme}) => ({
 
 export const PgTreeSelectionContext = React.createContext();
 
-export default function PgTreeView({ data = [], hasCheckbox = false, selectionChange = null}) {
+export default function PgTreeView({ data = [], hasCheckbox = false,
+  selectionChange = null, handleOnToggle = null, TreeNode = null }) {
```

Comment:
there is no need for `handleOnToggle`. You could simply add `...props` and pass those props to tree. That way you'll have more flexibility.
rename `TreeNode` to `NodeComponent` and in that case - rename `Node` to `DefaultNode`